src/components/general/ProtectedRoute/ProtectedRoute.jsx

```diff
@@ -3,14 +3,17 @@ import { Route, useHistory } from "react-router-dom";
 import { useAsync } from "react-async";
 import checkAuth from "../../../scripts/checkAuth";
 
-export default ({ component, path }) => {
+export default ({ component, path, adminLevel = false }) => {
   const history = useHistory();
   const { data, isPending } = useAsync({ promiseFn: checkAuth });
 
   if (isPending) return "Loading...";
-  if (data.isAuthenticated === false) {
-     history.push("/");
-     return;
+  if (
+    data.isAuthenticated === false ||
+    (adminLevel !== data.admin && adminLevel === true)
```

Comment:
We could refactor this line to be clearer:
```
    (adminLevel !== data.admin && adminLevel === true)
```

You care about whether `adminLevel` is `true` AND `data.admin` is `false`, so why not have:
```
(adminLevel === true && data.admin === false)
```